adventure.py

```diff
@@ -1,267 +1,514 @@
-
-
-
-# global-scope variables
-game_break = False # creating end condition for game screen loop
-player_inventory = [speed_potion, lesser_health_potion, greater_health_potion] # hard-coding a speed potion into the inventory for now
-#speed = 1 # for speed potion; DO NOT SET TO ZERO FOR ANY REASON
-number_of_player_moves = 0 # count of player moves for effect duration
-moves_until_effect_expires = {
-    "speed": 0
-}
-
-
-### FINDING PLAYER SPAWN POINT ###
-player_pos = [1,2] # creating player coordinate storage
-x = 0 # easy access to player position indices
-y = 1
-spawn_row = world.row(2)
-i = int(ceil(dim/3))
-for item in spawn_row: # finding empty space in first row for player to spawn
-    if item == " ":
-        player_pos[x] = i
-        break
-    i += 1
-world.mod_char(player_pos[x],2,colorama.Fore.WHITE + "+" if with_colors else "+") # marking origin on map
-world.print_tile() # printing the world for the first time
-
-"""
-functions for motion
-"""
-stored_tile = [colorama.Fore.WHITE + "O" if with_colors else "O"] # stores the tile the player is currently on (initial value will mark origin
-
-def reset_pos(): # resets after motion the tile that the player was on
-    world.mod_char(player_pos[x],player_pos[y],stored_tile[0])
-
-def detect_collision(coordinate,direction):
-    collision_output = []
-    if coordinate == "x":
-        for dict_element in world.tile_elements:
-            if world.char(player_pos[x]+direction,player_pos[y]) == dict_element["character"] and dict_element["is_viable"] == False:
-                collision_output.append(True)
-                collision_output.append(dict_element["name"])
-                return collision_output
-                #return True
-
-        collision_output.append(False)
-        return collision_output
-                #return False
-    elif coordinate == "y":
-        for dict_element in world.tile_elements:
-            if world.char(player_pos[x],player_pos[y]+direction) == dict_element["character"] and dict_element["is_viable"] == False:
-                collision_output.append(True)
-                collision_output.append(dict_element["name"])
-                return collision_output
-                #return True
-
-        collision_output.append(False)
-        return collision_output
-                #return False
-
-
-accepted_motions = [["w","a","s","d"],["2w","2a","2s","2d"]]
-def player_move(motion):
-    global number_of_player_moves
-    global moves_until_effect_expires
-    number_of_player_moves += 1 # upping the count of player moves by one
-
-    # making speed timer count down
-    if player.speed > 1:
-        if moves_until_effect_expires["speed"] == 0:
-            player.speed -= 1
+    dim = int(input("Tile dimension?\n"))  # getting world dimensions from user
+    name = input("Please enter your name:  ")
+
+    if platform.system() == "Windows" or platform.system() == "Linux":  # option to turn off colors to improve performance
```

Comment:
need an `else` statement aligned with first `if` so that `platform.system() != "Windows"` or `"Linux"` then `with_colors` is by default assigned to `False`

Reply:
Addressed. See commit `1b4acd7`.